internal/handler/msg_saslstart.go

```diff
@@ -61,25 +65,39 @@ func (h *Handler) MsgSASLStart(ctx context.Context, msg *wire.OpMsg) (*wire.OpMs
 			return nil, err
 		}
 
+		conninfo.Get(ctx).SetAuth(username, password)
+
+		var emptyPayload types.Binary
+		must.NoError(reply.SetSections(wire.MakeOpMsgSection(
+			must.NotFail(types.NewDocument(
+				"conversationId", int32(1),
+				"done", true,
+				"payload", emptyPayload,
+				"ok", float64(1),
+			)),
+		)))
+
+	case "SCRAM-SHA-256":
+		response, err := h.saslStartSCRAMSHA256(ctx, document)
+		if err != nil {
+			return nil, err
+		}
+
+		must.NoError(reply.SetSections(wire.MakeOpMsgSection(
+			must.NotFail(types.NewDocument(
+				"ok", float64(1),
+				"conversationId", int32(1),
```

Comment:
The `conversationId` doesn't seem to change for the duration of a conversation. See here for a [definition](https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#L431).

Reply:
It seems strange to have a field that never changes. Perhaps, conversationId is used by other authentication mechanism and SCRAM-SHA-256 does not use it, I'm not sure how it works

Reply:
It seems to indicate some state that's remembered during the conversation, but it doesn't make a lot of sense to me either.

> conversationId: the conversation identifier. This will need to be remembered and used for the duration of the conversation.